database/models/user.model.ts

```diff
@@ -26,6 +40,44 @@ const UserSchema = new Schema(
       type: String,
       required: false,
     },
+    learningGoals: {
+      type: [String],
+      required: false,
+    },
+    role: {
+      type: [Schema.Types.ObjectId],
+      required: false,
+      default: "user",
+      ref: "Role",
+    },
```

Comment:
Everything in this draft makes sense- except this portion here.

`role` looks like it's going to be an array of objectIds, but its default value is "user".

Reply:
oopsie. good catch